### PR TITLE
RFC: Feat/reconcile and account creation event

### PR DESCRIPTION
**Pull request by buckie**
Latest iteration, I'll add the tests and migration to `ledger-v2` etc once we agree on the semantics.

### CODE REVIEW

File: pact/ledger.pact
Comment (ggobugi27) [inline pact/ledger.pact:72]: `supply` doesn't seem necessary here, since it's always 0.0. What about we emit `TOKEN` instead of `SUPPLY` in 
https://github.com/kadena-io/marmalade/blob/1f6a52ea5fae517ca6f290580cbf7b9b6204e823/pact/ledger.pact#L418 
and remove SUPPLY?

File: pact/ledger.pact
Comment (ggobugi27) [inline pact/ledger.pact:390]: Also add `ACCOUNT_GUARD` here? New accounts are created at mint / transfer-create / offer / buy

File: pact/ledger.pact
Comment (ggobugi27) [inline pact/ledger.pact:379]: Our use of `-1.0` also emits `RECONCILE` of the receiver as `-1.0`,  so use 0.0 ? Does 0.0 make any difference?

**Review by ggobugi27 — COMMENTED** [review]
RECONCILE looks great!

File: pact/ledger.pact
Comment (buckie) [reply pact/ledger.pact:72]: Mostly future-proofing for the frontend, we may allow a newly created token to have a non-zero supply later on and it's nice to have a fully populated (to the default) state. I'd keep supply, as it's much more common. `TOKEN` is basically create account for tokens

File: pact/ledger.pact
Comment (buckie) [reply pact/ledger.pact:379]: good catch

File: pact/ledger.pact
Comment (buckie) [reply pact/ledger.pact:390]: I didn't realize that, thanks!

Comment (buckie) [reply pact/ledger.pact:390]: nah, I was just tired. Thanks for spotting the bug.

**Comment by ggobugi27** [comment]
Another comment about types: 
`transfer`, `transfer-create` end with `RECONCILE` events which is of type `bool`, so return type need to be updated to `bool`
`credit-account` also needs similar change to be typed to `object:{receiver-balance-change}`

Otherwise, I think it's good with update to `poly-fungible-v3` and tests

**Comment by ggobugi27** [comment]
- poly-fungible-v3 : RECONCILE, ACCOUNT_GUARD, sender-balance-change, receiver-balance-change, and changes type of functions that touch table to bool
- add yaml
- update repl tests

Locally, repl tests succeed... still figuring out why CI fails

**Review by buckie — COMMENTED** [review]
Lol, I can't approve this PR. I have no comment on the reviews, lgtm

**Review by ggobugi27 — APPROVED** [review]

**Pull request merged**
